fog/ledger/server/tests/router_integration.rs

```diff
@@ -254,74 +230,44 @@ fn create_router_client(
 
 fn create_env(
     config: TestEnvironmentConfig,
-    blocks_config: BlockConfig,
     grpc_env: Arc<grpcio::Environment>,
     logger: Logger,
 ) -> TestEnvironment {
-    let mut shards = vec![];
-    let mut stores = vec![];
-    let mut tempdirs = vec![];
-    for shard in config.shards.iter() {
-        for store in shard.stores.iter() {
-            let watcher_db_dir =
-                TempDir::new().expect("Couldn't create temporary path for watcher DB");
-            let ledger_db_dir =
-                TempDir::new().expect("Couldn't create temporary path for ledger DB");
-            stores.push(create_store(
-                store,
-                &blocks_config,
-                shard.block_range.clone(),
-                watcher_db_dir.path(),
-                ledger_db_dir.path(),
-                logger.clone(),
-            ));
-            tempdirs.push(watcher_db_dir);
-            tempdirs.push(ledger_db_dir);
-        }
+    let watcher_db_dir = TempDir::new().expect("Couldn't create temporary path for watcher DB");
+    let ledger_db_dir = TempDir::new().expect("Couldn't create temporary path for ledger DB");
+    let ledger = recreate_ledger_db(ledger_db_dir.path());
+    let watcher = setup_watcher_db(watcher_db_dir.path().to_path_buf(), logger.clone());
+    let mut block_provider = LocalBlockProvider::new(ledger, watcher);
+    seed_block_provider(&mut block_provider);
```

Comment:
Consolidated the block provider and only seed it with the genesis block here so that a follow on PR can add one block at a time and verify the state of the system after each block